### ISSUE TITLE
may I ask why you don't use pytorch nn.transformer directly

### ISSUE THREAD

**Issue by hyuan-kamuda** [olof98johansson/StockPrediction#6]
Thanks your code sharing!
I am try to learn how to use transformer for stock prediction( like using past 5 days trading data to forecast next day's open price,close price, volume....).
When I read your transformer code, it seems you manually implement a transformer NN instead of using pytorch's nn.transformer, why? 
I think we can set parameters below for it by using this built-in model architecture, right?

torch.nn.Transformer(d_model=512, nhead=8, num_encoder_layers=6, num_decoder_layers=6, dim_feedforward=2048, dropout=0.1, activation=<function relu>, custom_encoder=None, custom_decoder=None, layer_norm_eps=1e-05, batch_first=False, norm_first=False, device=None, dtype=None)

d_model – the number of expected features in the encoder/decoder inputs (default=512).

nhead – the number of heads in the multiheadattention models (default=8).

num_encoder_layers – the number of sub-encoder-layers in the encoder (default=6).

num_decoder_layers – the number of sub-decoder-layers in the decoder (default=6).

dim_feedforward – the dimension of the feedforward network model (default=2048).

dropout – the dropout value (default=0.1).

activation – the activation function of encoder/decoder intermediate layer, can be a string (“relu” or “gelu”) or a unary callable. Default: relu

custom_encoder – custom encoder (default=None).

custom_decoder – custom decoder (default=None).

layer_norm_eps – the eps value in layer normalization components (default=1e-5).

batch_first – If True, then the input and output tensors are provided as (batch, seq, feature). Default: False (seq, batch, feature).

norm_first – if True, encoder and decoder layers will perform LayerNorms before other attention and feedforward operations, otherwise after. Default: False (after).
